### DISCUSSION TITLE
Language Localization

### DISCUSSION THREAD

**Discussion by thegladscientist** [mozilla/hubs#3073] [General Questions & Discussion] [answered]
Does anyone know about language localization? I would like a hubs page to appear in Japanese by default, and even if it says default browser language in the settings, and a browser is set to Japanese it switches it to English, with only Chinese and i think French as choices in the menu. Does anyone have experience in doing something like this?

**Comment by johnshaughnessy** [comment, marked as answer]
Additional locales can be added by following the instructions here : https://github.com/mozilla/hubs/tree/master/src/assets/locales#localization
There is not Japanese language support in Hubs yet, but may be added soon. If someone submits a `ja.json` translation file before we get to it, we would happily merge it in.

**Reply by thegladscientist** [reply]
Thanks so much! I am working on a ja.json now, and will send it over soon!

**Reply by thegladscientist** [reply]
@johnshaughnessy I have a translation file for Japanese ready! 👍  

I couldn't attach a json here on GitHub so am putting a link to download on Dropbox (hope thats alright!) https://www.dropbox.com/s/p9lxgunyj9pi8kv/jp.json?dl=0

**Reply by johnshaughnessy** [reply]
Thanks @thegladscientist - I've opened a pull request using the translations file you uploaded to dropbox. https://github.com/mozilla/hubs/pull/3102

**Comment by tttigo** [comment]
Hello, I want to translate hubs to brazilian portuguese, how exactly should I do this? And where can I submit it? I'm newbie on this.

Cheers

**Reply by thegladscientist** [reply]
It looks relatively easy (albeit time consuming) to do using the link that @johnshaughnessy provided. You can basically take the en.json file from that page, and translate the text for each field for Portuguese. I copied and pasted into Google translate, and then am correcting where necessary for the fields that didn't translate well. Hopefully that helps you get started!

**Reply by tttigo** [reply]
Thanks!

Got it! Just submited a pull request with it!

**Reply by thegladscientist** [reply]
Hey @tttigo what is the best way to submit a pull request with it? I haven't worked much with big public repos like this so apologies for the question.

**Reply by johnshaughnessy** [reply]
Hi @thegladscientist - I went ahead and opened a pull request with your translation file : ([Add Japanese language option](https://github.com/mozilla/hubs/pull/3102))

Our [existing guidelines](https://hubs.mozilla.com/docs/contributing.html#-code-contributions) for contributing code do not explain how to use git and github in any detail, so I've added an [issue](https://github.com/mozilla/hubs/issues/3103) to our backlog suggesting that we add some links to documentation for git and github.  

In general, we would like to allow people to submit translations without needing them to know git / github, but have not yet set anything up for that yet. Thanks again for sharing the translations!

**Reply by thegladscientist** [reply]
Okay, cool, thank you very much! Happy to help the Hubs community in this little way!
